Examples/Algorithms/TrackFitting/src/GlobalChiSquareFitterFunction.cpp

```diff
@@ -0,0 +1,166 @@
+using DirectPropagator = Acts::Propagator<Stepper, Acts::DirectNavigator>;
+using DirectFitter =
+    Acts::KalmanFitter<DirectPropagator, Acts::VectorMultiTrajectory>;
+
+using TrackContainer =
+    Acts::TrackContainer<Acts::VectorTrackContainer,
+                         Acts::VectorMultiTrajectory, std::shared_ptr>;
+
+using namespace ActsExamples;
+
+struct GlobalChiSquareFitterFunctionImpl final : public TrackFitterFunction {
+  Fitter fitter;
+  DirectFitter directFitter;
+
+  bool multipleScattering = false;
+  bool energyLoss = false;
+  Acts::FreeToBoundCorrection freeToBoundCorrection;
+
+  IndexSourceLink::SurfaceAccessor m_slSurfaceAccessor;
+
+  GlobalChiSquareFitterFunctionImpl(Fitter&& f, DirectFitter&& df,
+                                    const Acts::TrackingGeometry& trkGeo)
+      : fitter(std::move(f)),
+        directFitter(std::move(df)),
+        m_slSurfaceAccessor{trkGeo} {}
+
+  template <typename calibrator_t>
+  auto makeGx2fOptions(const GeneralFitterOptions& options,
+                       const calibrator_t& calibrator) const {
+    Acts::Experimental::Gx2FitterExtensions<Acts::VectorMultiTrajectory>
+        extensions;
+    extensions.calibrator.connect<&calibrator_t::calibrate>(&calibrator);
+
+    extensions.surfaceAccessor
+        .connect<&IndexSourceLink::SurfaceAccessor::operator()>(
+            &m_slSurfaceAccessor);
+
+    const Acts::Experimental::Gx2FitterOptions gx2fOptions(
+        options.geoContext, options.magFieldContext, options.calibrationContext,
+        extensions, options.propOptions, &(*options.referenceSurface),
+        multipleScattering, energyLoss, freeToBoundCorrection, 5);
+
+    return gx2fOptions;
+  }
+
+  TrackFitterResult operator()(const std::vector<Acts::SourceLink>& sourceLinks,
+                               const TrackParameters& initialParameters,
+                               const GeneralFitterOptions& options,
+                               const MeasurementCalibratorAdapter& calibrator,
+                               TrackContainer& tracks) const override {
+    const auto gx2fOptions = makeGx2fOptions(options, calibrator);
+    return fitter.fit(sourceLinks.begin(), sourceLinks.end(), initialParameters,
+                      gx2fOptions, tracks);
+  }
+
+  // TODO create operator() for directNavigator
+  // We need a placeholder for the directNavigator overload, to safely compile.
+  TrackFitterResult operator()(
+      const std::vector<Acts::SourceLink>& sourceLinks,
+      const TrackParameters& initialParameters,
+      const GeneralFitterOptions& options,
+      const RefittingCalibrator& calibrator,
+      const std::vector<const Acts::Surface*>& surfaceSequence,
+      TrackContainer& tracks) const override {
+    const auto gx2fOptions = makeGx2fOptions(options, calibrator);
+    throw std::runtime_error(
+        "direct navigation with GX2 fitter not yet implemented");
+    (void)surfaceSequence;
```

Comment:
this is now unreachable code, I wonder why the compiler does not complain with a warning... probably I would remove this (as well as the `gx2fOption` above)